### PR TITLE
Add support for matching licenses in Cargo.toml

### PR DESCRIPTION
**Pull request by andrew**
Cargo is the package manager for Rust, manifest files are written in toml

### CODE REVIEW

**Comment by benbalter** [comment]
@andrew Wow. Thanks for this. Beautifully written. Tried to review, but not a single comment. 🎉

**Comment by andrew** [comment]
@benbalter awesome, in that case I'll send a few more prs over for other package managers too, we're looking at using licensee in https://libraries.io this year.

**Comment by benbalter** [comment]
@andrew That'd be awesome.  More than glad to review PRs. Feedback (e.g. on the API) also welcome. Just invited you to be a collaborator in case the branching model is easier.

**Comment by andrew** [comment]
Awesome, thanks!

File: lib/licensee/matchers/cargo.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module Licensee
+  module Matchers
+    class Cargo < Licensee::Matchers::Package
+      LICENSE_REGEX = /
+        ^\s*[\'\"]?license[\'\"]?\s*=\s*[\'\"]([a-z\-0-9\._]+)[\'\"]\s*
```
Comment (wking) [inline lib/licensee/matchers/cargo.rb]: You may also want to allow internal parens.  They are not yet supported by crates.io (withoutboats/license-exprs#3, and I'm documenting that limitation in rust-lang/cargo#4898), but hopefully they'll be supported at some point.  And I don't see a problem with allowing them in this regex now.

Comment (wking) [reply lib/licensee/matchers/cargo.rb]: > You may also want to allow internal parens…

I've filed support for this in #261, although Licensee will still return `other` for those cases.

**Pull request merged**